quil-py/quil/instructions/__init__.pyi

```diff
@@ -269,13 +273,12 @@ class ArithmeticOperator(Enum):
     Multiply = "Multiply"
 
 class Arithmetic:
-    @staticmethod
     def __new__(
         cls,
         operator: ArithmeticOperator,
         destination: ArithmeticOperand,
         source: ArithmeticOperand,
-    ) -> "Arithmetic": ...
+    ) -> Self: ...
```

Comment:
This fixes an issue where `mypy` would expect a sub class to return an instance of itself even when leaning on it's parent class for initialization. For exampe:

```py
class Base:
    def __new__(cls) -> "Base": ...

class Sub(Base):
    def __new__(cls) -> "Sub":
        return super().__new__(cls)  # mypy complains that the function should return `Sub` but it returns `Base`
```

The above is valid code, an instance of `Sub()` would work as you would expect `Sub` to work, and calling `super().__new__` to initialize a subclass is what the [Python docs recommend](https://docs.python.org/3/reference/datamodel.html?highlight=__new__#object.__new__).

The [`Self` type](https://peps.python.org/pep-0673/) is essentially a shortcut for a generic `TypeVar` bounded by the parent class, meaning that a return type of `Self` can be used in a child class without violating any type rules.